### PR TITLE
Update release process and prepare 1.8 development cycle

### PR DESCRIPTION
**Pull request by aanm**
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/10044)

### CODE REVIEW

**Comment by aanm** [comment]
test-docs-please

**Comment by aanm** [comment]
test-docs-please

**Comment by aanm** [comment]
test-docs-please

File: Documentation/contributing/release/feature.rst [hunk 1]
```diff
@@ -48,15 +56,45 @@ On Freeze date
...
    #. ``needs-backport/1.2``
 
+
+#. Checkout to master and update the ``.github/cilium-actions.yml`` to have
+   all the necessary configurations for the backport of the new ``vX.Y`` branch.
+
+   ::
+
+       git checkout master
+       git checkout -b pr/master-cilium-actions-update
```
Comment (joestringer) [inline Documentation/contributing/release/feature.rst]: nit: The above two could be combined into `git checkout -b pr/master-cilium-actions-update master`

(The last `master` could even be `origin/master`)

**Review by joestringer — APPROVED** [review]

File: Documentation/contributing/release/feature.rst [hunk 2]
```diff
@@ -48,15 +56,45 @@ On Freeze date
...
+#. Continue with the next step only after the previous steps are merged into
+   master.
+
+#. Mark all open PRs with ``needs-backport/x.y`` that have the milestone ``x.y``
+
+#. Change the VERSION file to contain the next ``rc`` version. For example,
+   if we are branching v1.2 and still in the RC phase we need to change the
+   VERSION file to contain the ``v1.2.0-rcX``
+
+#. Set the branch as "Active" in the readthedocs Admin page. (Replace ``v1.2``
```
Comment (aanm) [inline Documentation/contributing/release/feature.rst]: Change this option to be active but set it to Privacy level "private"

**Comment by aanm** [comment]
test-docs-please

**Pull request merged**